### PR TITLE
Warning icon displays when the users budget is over 75%, a modal pops up when the icon is tapped

### LINKED ISSUE #41: FR17. Warning modal
| FR17. Warning modal and toast notification |
|-----------------------|
| Goal: FR17 Conditionally shows a modal when a user is close to being over budget in the expenses page|
| Stakeholders: Secondary and Post-Secondary Students, Finance beginners |
| Description: When the user is approaching their budget limit, a pop up in the form of a modal will appear on the expenses page. |
| Origin: Team meeting on 3/23 |
| Version: 1.0 |
| Date: 03/23/2023 |
| Priority: 2|

### PR DESCRIPTION
**Pull request by amuwhao46**

### CODE REVIEW

**Review by mxsingh — APPROVED** [review]
Looks good!

File: lib/ExpensesPage/warningModal.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        children: <Widget>[
+          const Icon(Icons.warning_amber_rounded, size: 50.0),
+          const Text(
+            'Attention!',
+            style: TextStyle(fontWeight: FontWeight.bold, fontSize: 20),
+          ),
+          const Padding(
+            padding: EdgeInsets.all(16.0),
+            child: Text(
+                'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut faucibus pulvinar elementum integer enim. Ante metus dictum at tempor commodo ullamcorper a. Viverra justo nec ultrices dui sapien eget mi proin sed. Semper auctor neque vitae tempus. Maecenas pharetra convallis posuere morbi leo urna molestie at elementum. Volutpat commodo sed egestas egestas fringilla phasellus. Id ornare arcu odio ut sem nulla pharetra diam. Vel risus commodo viverra maecenas accumsan lacus vel facilisis volutpat. Mauris vitae ultricies leo integer malesuada nunc vel. Ullamcorper velit sed ullamcorper morbi.'),
```
Comment (Ahmed-Alagbari) [inline lib/ExpensesPage/warningModal.dart:29]: is this just placeholder text?

File: lib/ExpensesPage/expensesPage.dart [hunk 2]
```diff
@@ -28,7 +29,34 @@ class ExpensesPage extends StatelessWidget {
           mainAxisSize: MainAxisSize.min,
           mainAxisAlignment: MainAxisAlignment.start,
           children: [
-            Row(
+            (percentHelper(progressAmt, totalProgress) >= .75)
```
Comment (Ahmed-Alagbari) [inline lib/ExpensesPage/expensesPage.dart:32]: so does the modal show up when a budget is at 75% or lower?

File: lib/ExpensesPage/warningModal.dart (hunk 1 above)
Comment (amuwhao46) [reply lib/ExpensesPage/warningModal.dart:29]: Yeah it's place holder text, I'll update it in the future

File: lib/ExpensesPage/expensesPage.dart (hunk 2 above)
Comment (amuwhao46) [reply lib/ExpensesPage/expensesPage.dart:32]: No the warning icon shows up only when the budget is 75% or higher, when the icon is visible, you can tap it, then the modal will pop up

**Review by Ahmed-Alagbari — APPROVED** [review]
Looks good!

**Pull request merged**